### DISCUSSION TITLE
Netmiko Exception as Unknown error

### DISCUSSION THREAD

**Discussion by akshayajai7** [ktbyers/netmiko#2348] [Q&A]
Hey all, 
I tried deploying some configurations(Configuring Vsileft and vasiright interfaces) to a Cisco router . The first time deployment raised the exception in Netmiko which is :
    except Exception as unknown_error:
        print("Some other error: " + str(unknown_error))
 And the error was: Some other error: [Errno None] Unable to connect to port 22 in x.x.x.x
I tried a second time, it  didn't raise the same exception and deployment went through . 
What is the reason for this kind of behavior? Any help would really be appreciated.    
```
def connect_to_the_device(device_hostname):
    device = {
        'device_type': 'cisco_ios',
        'host': str(device_hostname),
        'username': str(user_name),
        'key_file': str(key_file)
    }
    try:
        ssh_conn = ConnectHandler(**device)
        scp_conn = SCPConn(ssh_conn)
        print(f"connected to the device: {device_hostname}")
        return ssh_conn, scp_conn

    except AuthenticationException:
        print("Authentication failure: " + str(device_hostname))
        sys.exit(1)
    except NetMikoTimeoutException:
        print("Timeout to device: " + str(device_hostname))
        sys.exit(1)
    except EOFError:
        print("End of file while attempting device " + str(device_hostname))
        sys.exit(1)
    except SSHException:
        print("SSH Issue. Are you sure SSH is enabled? " + str(device_hostname))
        sys.exit(1)
    except Exception as unknown_error:
        print("Some other error: " + str(unknown_error))
        sys.exit(1) 
```

**Comment by ktbyers** [comment]
Can you get rid of all your exception handling and post the original stack trace so I can see the actual error?

**Reply by akshayajai7** [reply]
Hey @ktbyers Sorry for the delayed response, since I was not actually able to replicate the same issue that I had. As you suggested, I got rid of all exceptions but still I could only get [Errno None] Unable to connect to port 22 on x.x.x.x
So , not sure if it is something on the device or Netmiko. I see this error not all time but sometimes.

**Reply by ktbyers** [reply]
Okay, do you have the full exception stack trace and message from when it fails?


Kirk
